src/Psalm/Internal/Analyzer/ClassAnalyzer.php

```diff
@@ -271,6 +272,22 @@ public function analyze(
             );
         }
 
+        $classUnion = new Union([new TNamedObject($fq_class_name)]);
+        foreach ($storage->parent_classes + $storage->direct_class_interfaces as $parent_class) {
```

Comment:
This makes me wonder whether the following should be flagged or not:

```php
/** @psalm-inheritors Whatever */
interface I {}
interface IC extends I {}
```

Reply:
I'm of the belief extending the interface should be explicitly stated if the annotation is used. Updated to analyse interfaces too.